src/coord/src/catalog/storage.rs

```diff
@@ -438,6 +438,33 @@ impl Transaction<'_> {
         }
     }
 
+    fn assert_timestamp_bindings(&self, source_id: &GlobalId) {
+        let bindings: Vec<(PartitionId, Timestamp, MzOffset)> = self.inner
```

Comment:
Any reason to duplicate this code instead of just calling `load_timestamp_bindings(source_id).unwrap()`?